symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/AuthenticatorFactory.java

```diff
@@ -1,214 +1,42 @@
 package com.symphony.bdk.core.auth;
 
-import static com.symphony.bdk.core.util.DeprecationLogger.logDeprecation;
-import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
-
 import com.symphony.bdk.core.auth.exception.AuthInitializationException;
-import com.symphony.bdk.core.auth.impl.BotAuthenticatorCertImpl;
-import com.symphony.bdk.core.auth.impl.BotAuthenticatorRsaImpl;
-import com.symphony.bdk.core.auth.impl.ExtensionAppAuthenticatorCertImpl;
-import com.symphony.bdk.core.auth.impl.ExtensionAppAuthenticatorRsaImpl;
-import com.symphony.bdk.core.auth.impl.OboAuthenticatorCertImpl;
-import com.symphony.bdk.core.auth.impl.OboAuthenticatorRsaImpl;
-import com.symphony.bdk.core.auth.jwt.JwtHelper;
-import com.symphony.bdk.core.client.ApiClientFactory;
-import com.symphony.bdk.core.config.model.BdkAuthenticationConfig;
-import com.symphony.bdk.core.config.model.BdkConfig;
 
-import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.io.IOUtils;
 import org.apiguardian.api.API;
 
-import java.io.FileInputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
-import java.security.GeneralSecurityException;
-import java.security.PrivateKey;
-
 import javax.annotation.Nonnull;
 
 /**
- * Factory class that provides new instances for the main authenticators :
+ * Interface for authentication factory class to provide new instances for the main authenticators :
  * <ul>
  *   <li>{@link BotAuthenticator} : to authenticate the main Bot service account</li>
  *   <li>{@link OboAuthenticator} : to perform on-behalf-of authentication</li>
  * </ul>
  */
-@Slf4j
 @API(status = API.Status.STABLE)
-public class AuthenticatorFactory {
-
-  private final BdkConfig config;
-  private final ApiClientFactory apiClientFactory;
-
-  public AuthenticatorFactory(@Nonnull BdkConfig bdkConfig, @Nonnull ApiClientFactory apiClientFactory) {
-    this.config = bdkConfig;
-    this.apiClientFactory = apiClientFactory;
-  }
+public interface AuthenticatorFactory {
```

Comment:
isn't it going to be a breaking change? (as it is public and marked as stable)

Reply:
an alternative would be to let them subclass the class and having the constructor's parameter would be enough

we might want to check if they are using the spring boot integration too and if we need it to support it there

Reply:
ok so I will remove the interface and only implement the new constructor then

About spring boot I saw you left a comment in the ticket, thanks!